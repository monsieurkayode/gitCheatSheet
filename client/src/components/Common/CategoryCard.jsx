import React from 'react';
import styled from 'styled-components';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

const Container = styled.div`
  height: 64px;
  flex: 1 1 23%;
  min-width: 160px;
  background: #668090;
  border-radius: 4px;
  margin: 5px;
  padding: 0 16px;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  text-transform: uppercase;
  font-family: Prompt, sans-serif;
  font-size: 13px;
  cursor: pointer;
  user-select: none;
  color: #fff;

  p {
    width: 100%;
  }

  .fiber {
    vertical-align: sub;
    font-size: 14px;
    color: #93d82a;
  }

  &:hover {
    background: #546975;
  }

  @media only screen and (max-width: 424px) {
    flex: 1 1 98%;
  }
`;

const CategoryCard = ({ category, history }) => (
  <Container
    onClick={() => history.push(`/cheats/${category.replace(/[ ]/g, '-')}`)}
    key={category}
    className="category"
  >
    <p>
      {category}&nbsp;&nbsp;
      <i className="material-icons fiber">fiber_manual_record</i>
    </p>
  </Container>
);

CategoryCard.propTypes = {
  category: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export default withRouter(CategoryCard);
