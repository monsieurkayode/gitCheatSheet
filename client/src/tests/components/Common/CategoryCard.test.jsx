import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import CategoryCard from '../../../components/Common/CategoryCard';

const setup = () => {
  const props = {
    category: 'Branch',
    history: {
      push: jest.fn()
    }
  };

  const wrapper = mount(
    <BrowserRouter>
      <CategoryCard {...props} />
    </BrowserRouter>
  );

  return { wrapper, props };
};

describe('Category Card Component', () => {
  const { wrapper } = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the history push method', () => {
    wrapper.find('.category').first().simulate('click');
  });
});
