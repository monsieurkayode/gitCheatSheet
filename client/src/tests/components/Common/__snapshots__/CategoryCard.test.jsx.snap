// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Card Component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(CategoryCard)
      category="Branch"
      history={
        Object {
          "push": [MockFunction],
        }
      }
    >
      <Route>
        <CategoryCard
          category="Branch"
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <styled.div
            className="category"
            key="Branch"
            onClick={[Function]}
          >
            <StyledComponent
              className="category"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "bVUSMF",
                    "rules": Array [
                      "
  height: 64px;
  flex: 1 1 23%;
  min-width: 160px;
  background: #668090;
  border-radius: 4px;
  margin: 5px;
  padding: 0 16px;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  text-transform: uppercase;
  font-family: Prompt, sans-serif;
  font-size: 13px;
  cursor: pointer;
  user-select: none;
  color: #fff;

  p {
    width: 100%;
  }

  .fiber {
    vertical-align: sub;
    font-size: 14px;
    color: #93d82a;
  }

  &:hover {
    background: #546975;
  }

  @media only screen and (max-width: 424px) {
    flex: 1 1 98%;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <div
                className="category sc-bdVaJa bVUSMF"
                onClick={[Function]}
              >
                <p>
                  Branch
                    
                  <i
                    className="material-icons fiber"
                  >
                    fiber_manual_record
                  </i>
                </p>
              </div>
            </StyledComponent>
          </styled.div>
        </CategoryCard>
      </Route>
    </withRouter(CategoryCard)>
  </Router>
</BrowserRouter>
`;
